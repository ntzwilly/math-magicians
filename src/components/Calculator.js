import React, { useState } from 'react';
import calculate from '../logic/calculate';

export default function Calculator() {
  const [state, setState] = useState({});

  const onClickHandler = (event) => {
    setState((prevState) => calculate(prevState, event.target.value));
  };

  const displayResult = () => {
    const { total, next, operation } = state;
    const display = (total || '') + (operation || '') + (next || '');

    return display || '0';
  };

  return (
    <div className="container">
      <div className="result">
        <span>{displayResult()}</span>
      </div>
      <div className="elements">
        <button type="button" className="number" onClick={onClickHandler} value="AC">AC</button>
        <button type="button" className="number" onClick={onClickHandler} value="+/-">+/-</button>
        <button type="button" className="number" onClick={onClickHandler} value="%">%</button>
        <button type="button" className="operator" onClick={onClickHandler} value="รท">รท</button>
        <button type="button" className="number" onClick={onClickHandler} value="7">7</button>
        <button type="button" className="number" onClick={onClickHandler} value="8">8</button>
        <button type="button" className="number" onClick={onClickHandler} value="9">9</button>
        <button type="button" className="operator" onClick={onClickHandler} value="x">x</button>
        <button type="button" className="number" onClick={onClickHandler} value="4">4</button>
        <button type="button" className="number" onClick={onClickHandler} value="5">5</button>
        <button type="button" className="number" onClick={onClickHandler} value="6">6</button>
        <button type="button" className="operator" onClick={onClickHandler} value="-">-</button>
        <button type="button" className="number" onClick={onClickHandler} value="1">1</button>
        <button type="button" className="number" onClick={onClickHandler} value="2">2</button>
        <button type="button" className="number" onClick={onClickHandler} value="3">3</button>
        <button type="button" className="operator" onClick={onClickHandler} value="+">+</button>
        <button type="button" className="number btn-0" onClick={onClickHandler} value="0">0</button>
        <button type="button" className="number" onClick={onClickHandler} value=".">.</button>
        <button type="button" className="operator" onClick={onClickHandler} value="=">=</button>
      </div>
      <div className="elements">
        <button type="button" className="number" onClick={onClickHandler} value="AC">AC</button>
        <button type="button" className="number" onClick={onClickHandler} value="+/-">+/-</button>
        <button type="button" className="number" onClick={onClickHandler} value="%">%</button>
        <button type="button" className="operator" onClick={onClickHandler} value="รท">รท</button>
        <button type="button" className="number" onClick={onClickHandler} value="7">7</button>
        <button type="button" className="number" onClick={onClickHandler} value="8">8</button>
        <button type="button" className="number" onClick={onClickHandler} value="9">9</button>
        <button type="button" className="operator" onClick={onClickHandler} value="x">x</button>
        <button type="button" className="number" onClick={onClickHandler} value="4">4</button>
        <button type="button" className="number" onClick={onClickHandler} value="5">5</button>
        <button type="button" className="number" onClick={onClickHandler} value="6">6</button>
        <button type="button" className="operator" onClick={onClickHandler} value="-">-</button>
        <button type="button" className="number" onClick={onClickHandler} value="1">1</button>
        <button type="button" className="number" onClick={onClickHandler} value="2">2</button>
        <button type="button" className="number" onClick={onClickHandler} value="3">3</button>
        <button type="button" className="operator" onClick={onClickHandler} value="+">+</button>
        <button type="button" className="number btn-0" onClick={onClickHandler} value="0">0</button>
        <button type="button" className="number" onClick={onClickHandler} value=".">.</button>
        <button type="button" className="operator" onClick={onClickHandler} value="=">=</button>
      </div>
    </div>
  );
}
