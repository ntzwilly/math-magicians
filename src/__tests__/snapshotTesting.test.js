import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import Home from '../components/Home';

describe('Snapshot testing components', () => {
  it('Calculator snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
