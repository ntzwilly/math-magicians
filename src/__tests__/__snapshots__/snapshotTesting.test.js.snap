// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing components Calculator snapshot 1`] = `
<div
  className="calculator"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="container"
  >
    <div
      className="result"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="elements"
    >
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="%"
      >
        %
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="7"
      >
        7
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="8"
      >
        8
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="9"
      >
        9
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="x"
      >
        x
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="4"
      >
        4
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="5"
      >
        5
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="6"
      >
        6
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="-"
      >
        -
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="1"
      >
        1
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="2"
      >
        2
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="3"
      >
        3
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="+"
      >
        +
      </button>
      <button
        className="number btn-0"
        onClick={[Function]}
        type="button"
        value="0"
      >
        0
      </button>
      <button
        className="number"
        onClick={[Function]}
        type="button"
        value="."
      >
        .
      </button>
      <button
        className="operator"
        onClick={[Function]}
        type="button"
        value="="
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Snapshot testing components Home snapshot 1`] = `
<div>
  <div
    className="home-content"
  >
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
  </div>
</div>
`;
